extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php

```diff
@@ -295,221 +209,115 @@ private function checkIfSubscribedToWeeklyDigest( \User $user ) {
-				}
-
-				foreach ( $aDigest['pages'] as $aPageData ) {
-					// watchlist tracking, rt#33913
-					$url = $aPageData['title']->getFullURL( 's=dgdiff' . ( $aPageData['revisionId'] ? "&diff=" . $aPageData['revisionId'] . "&oldid=prev" : "" ) );
-
-					// plain email
-					$sDigests .= $url . "\n";
+	private function getDigestData( $userId ) {
 
-					// html email
-					if ( $usehtmlemail ) {
-						$pagename = $aPageData['title']->getArticleName();
-						$pagename = str_replace( '_', ' ', rawurldecode( $pagename ) );
-						$sDigestsHTML .= '<li><a href="' . $url . '">' . $pagename . "</a></li>\n";
-					}
+		$loop = 0;
+		$digestData = [];
+		foreach( $this->getWatchedPagesForUser( $userId ) as $watchedPage ) {
 
-					$iPagesCount++;
-				}
+			if ( $loop >= self::MAX_PAGES_PER_DIGEST ) {
+				break;
+			}
 
-				if ( $usehtmlemail ) {
-					$sDigestsHTML .= "</ul>\n<br/>\n";
-				}
+			$wikiaName = $this->getWikiaName( $watchedPage->gwa_city_id );
+			if ( empty( $wikiaName ) ) {
+				continue;
 			}
 
-			# blog comments
-			if ( !empty( $aDigest['blogs'] ) ) {
-				foreach ( $aDigest['blogs'] as $blogTitle => $blogComments ) {
-					# $countComments = ($blogComments['comments'] >= $blogComments['own_comments']) ? intval($blogComments['comments'] - $blogComments['own_comments']) : $blogComments['comments'];
-					$countComments = $blogComments['comments'];
-
-					$tracking_url = $blogComments['blogpage']->getFullURL( 's=dg' ); // watchlist tracking, rt#33913
-
-					$message = wfMsgReplaceArgs(
-						( $countComments != 0 ) ? $this->getLocalizedMsg( 'globalwatchlist-blog-page-title-comment', $oUserLanguage ) : "$1",
-						[
-							0 => $tracking_url, // send the ugly tracking url to the plain emails
-							1 => $countComments
-						]
-					);
-					$sDigestsBlogs .= $message . "\n";
-
-					if ( $usehtmlemail ) {
-						// for html emails, remake some things
-						$clean_url = $blogComments['blogpage']->getFullURL();
-						$clean_url = str_replace( '_', ' ', rawurldecode( $clean_url ) );
-						$message = wfMsgReplaceArgs(
-							( $countComments != 0 ) ? $this->getLocalizedMsg( 'globalwatchlist-blog-page-title-comment', $oUserLanguage ) : "$1",
-							[
-								0 => "<a href=\"{$tracking_url}\">" . $clean_url . "</a>", // but use the non-tracking one for html display
-								1 => $countComments
-							]
-						);
-						$sDigestsBlogsHTML .= $message . "<br/>\n";
-					}
-
-					$iBlogsCount++;
-				}
-				$sDigestsBlogs .= "\n";
+			$title = $this->getTitle( $watchedPage );
+			if ( !$title->exists() ) {
+				continue;
 			}
 
-			$sDigests .= "\n";
-		}
-		if ( $isDigestLimited ) {
-			$sDigests .= $this->getLocalizedMsg( 'globalwatchlist-see-more', $oUserLanguage ) . "\n";
-		}
-		$aEmailArgs = [
-			0 => ucfirst( $oUser->getName() ),
-			1 => ( $iPagesCount > 0 ) ? $sDigests : $this->getLocalizedMsg( 'globalwatchlist-no-page-found', $oUserLanguage ),
-			2 => ( $iBlogsCount > 0 ) ? $sDigestsBlogs : "",
```

Comment:
Oh god this is terrible; why explicitly call out the keys only to make them numbered indexes?!  Either make a normal array or make meaningful names ...